tests/AssetBank/FileUploaderTest.php

```diff
@@ -18,269 +22,390 @@ protected function setUp()
+        return new FulfilledPromise(
+            [
+                'test' => 'SuccessfulResponseUploadChunk'
+            ]
+        );
     }
 
     /**
-     * Builds a valid register chunk request.
+     * Builds a valid finalise api request.
      *
+     * @param string $fileId of the file to be uploaded returned by the prepare.
+     * @param string Path of the file to be uploaded.
+     * @param integer $fileSize of the file to be uploaded.
+     * @param integer $chunksCount denoting number of chunks in which the file is to be uploaded.
+     * @param string $fileSha256 digest of the file is to be uploaded.
+     * 
      * @return array The request params.
      */
-    private static function getRegisterChunkRequest()
+    private static function getFinaliseApiRequest($fileId, $filePath, $fileSize, $chunksCount, $fileSha256)
     {
         return [
             'POST',
-            'api/v4/upload/',
+            'v7/file_cmds/upload/' . $fileId . '/finalise_api',
             [
                 'form_params' => [
-                    'id' => 'fakeUploadId',
-                    'targetid' => 'fakeTargetid',
-                    'filename' => 's3_filename/p1',
-                    'chunkNumber' => 1,
+                    'fileName' => basename($filePath),
+                    'fileSize' => $fileSize,
+                    'chunksCount' => $chunksCount,
+                    'sha256' => $fileSha256,
+                    'intent' => "upload_main_uploader_asset"
                 ]
             ]
         ];
     }
 
     /**
-     * Returns a Fullfilled promise with a fake UploadId.
+     * Returns a fulfilled promise with the correlationId in the header for the finalise_api request.
      *
      * @return FulfilledPromise
      */
-    private static function getRegisterChunkResponse()
+    private static function getFinaliseApiResponse()
     {
-        return new FulfilledPromise('fakeUploadId');
+
+        return new FulfilledPromise(
+            new Psr7\Response(200, ['X-API-Correlation-ID' => 'TesterCorrelationId'], 'body string')
+        );
     }
 
     /**
-     * Builds a valid finalise upload request.
+     * Builds a valid save media request.
+     *
+     * @param string $fileId of the file to be uploaded returned by the prepare.
+     * @param string $filePath of the file to be uploaded.
      *
      * @return array The request params.
      */
-    private static function getFinaliseRequest()
+    private static function getSaveMediaRequest($fileId, $filePath, $brandId)
     {
         return [
             'POST',
-            'api/v4/upload/',
+            'api/v4/media/save/' . $fileId,
             [
                 'form_params' => [
-                    'id' => 'fakeUploadId',
-                    'targetid' => 'fakeTargetid',
-                    's3_filename' => 's3_filename/p1',
-                    'chunks' => 1,
+                    'filePath' => $filePath,
```

Comment:
Not sure why this param was once added to the request, but it is not part of the params listed in the API docs.